### PR TITLE
Add Task for Gradle builds

### PR DESCRIPTION
**Pull request by crshnburn**
- Create Gradle Task
- Add tests for gradle task
- Move first version into 0.1 directory

# Changes

Add a new task which runs a Gradle build which can then feed into other tasks.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by chmouel** [comment]
/ok-to-test

any chances to sign the CLA? thanks

**Comment by crshnburn** [comment]
> any chances to sign the CLA? thanks

CLA is now signed.

File: task/gradle/0.1/gradle.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    This Task can be used to run a Gradle build.
+
+  workspaces:
+    - name: source
+      description: The workspace consisting of the gradle project.
+  params:
+    - name: GRADLE_IMAGE
+      description: Gradle base image.
+      type: string
+      default: gradle:6.9.0-jdk8
```
Comment (vinamra28) [inline task/gradle/0.1/gradle.yaml:23]: can you please use the full qualified image name eg `docker.io/library/alpine:3.14@sha256:dea76d4b50c75125cada676a87ac23de2b7ba4374752c6f908253c3b839201d9` ?

File: task/gradle/OWNERS [hunk 2]
```diff
@@ -0,0 +1,4 @@
+approvers:
+- crshnburn
+reviewers:
+- crshnburn
```
Comment (vinamra28) [inline task/gradle/OWNERS:4]: to add yourself as the owner of this task, you need to sign the CLA first and then add yourself in the Tekton Org. Can you please do the needful?

**Review by vinamra28 — CHANGES_REQUESTED** [review]

File: task/gradle/OWNERS (hunk 2 above)
Comment (crshnburn) [reply task/gradle/OWNERS:4]: https://github.com/tektoncd/community/pull/621 raised to add myself to the org.

File: task/gradle/0.1/gradle.yaml (hunk 1 above)
Comment (crshnburn) [reply task/gradle/0.1/gradle.yaml]: Set image to `docker.io/library/gradle:6.9-jdk8@sha256:571312a1555f5a43305e3a73505c1786c335e8acc441495a396f054194327885`

**Comment by piyush-garg** [comment]
Can you add the example run in readme also

also please squash the commits

File: task/gradle/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Gradle
```
Comment (vinamra28) [inline task/gradle/README.md:1]: can you move this file to task/gradle/0.1/README.md

File: task/gradle/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+This Task can be used to run a Gradle build on a gradle project.
+
+## Install the Task
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/gradle/gradle.yaml
```
Comment (vinamra28) [inline task/gradle/README.md:8]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/gradle/0.1/gradle.yaml
```

**Comment by crshnburn** [comment]
@piyush-garg Example added and commits squashed

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**